DFReader.py

```diff
@@ -732,6 +833,8 @@ def _add_msg(self, m):
                 self.mav_type = mavutil.mavlink.MAV_TYPE_SUBMARINE
             elif m.Message.find("Blimp") != -1:
                 self.mav_type = mavutil.mavlink.MAV_TYPE_AIRSHIP
+            # Reset the meta data, when mav type is updated
+            self.metadata.reset()
```

Comment:
should only do it when mav_type changes

Reply:
I did some checks, and as speculated, this doesn't appear to be needed at all..... init_arrays, called by constructor will have set the mav_type for the file before anyone can try accessing any metadata. Therefore I propose to remove the lines.